### ISSUE TITLE
Reorder files in Zenodo archiver

### ISSUE THREAD

**Issue by e-belfer** [catalyst-cooperative/pudl#3006]
**Is your feature request related to a problem? Please describe.**
In Zenodo, files are ordered in the record using arbitrary ordering. Datapackage.json uses this ordering, so this also shows up in the list of resources in `datapackage.json`. Not technically a bug, but also not super ergonomic if someone is trying to look at the metadata.

Unfortunately, as noted in [this PR](https://github.com/catalyst-cooperative/pudl-archiver/pull/184) the `creators` field response in the Zenodo API != the `creators` field expected format (e.g. the `person_or_org` field is not returned, among other changes). This means you can't GET a deposition and PUT its metadata back. Updating a draft record with one field (e.g. the order field for the files dictionary) also seems to wipe all the rest of the inputted values for the metadata. This very annoying behavior that means we should avoid updating records by any means other than linking once they're created, unless we write a method to digest GET data and reformat it into the PUT request, or carry the originally input metadata by some other means. Files are downloaded asynchronously and uploaded as they're downloaded one by one, which is desired from the perspective of robustness but makes ordering them impossible until after all downloads are done.

**Describe the solution you'd like**
Some way to order files - either by using the `order` field in the `metadata` once all keys are known, or by uploading files in order. This should carry over to the datapackage.

**Describe alternatives you've considered**
Simply PUT, PUT'ing the `order` once all files are uploaded to the deposition wipes existing metadata. One option here is to do some cartwheeling to reupload all the metadata but it's not immediately obvious to me what's the best way forward here.
